sonarqube/content.md

```diff
@@ -30,68 +45,145 @@ $ mvn sonar:sonar -Dsonar.host.url=http://$(boot2docker ip):9000
+3. Initialize the new `SQ_HOME` with `--init`:
+```console
+$ docker run --rm \
+    -v $SQ_HOME/conf:/opt/sonarqube/conf \
+    -v $SQ_HOME/extensions:/opt/sonarqube/extensions \
+    -v $SQ_HOME/data:/opt/sonarqube/data \
+    sonarqube --init  
+```
+
+4. Follow the normal upgrade procedure https://docs.sonarqube.org/latest/setup/upgrading/
+  - Manually install the non-default plugins
+  - Update the contents of sonar.properties and wrapper.conf
+  - etc.  
+  
+5. Stop and remove the sonarqube container (restart is not enough as the env variable are not evaluated only during the first run, not during the restart):
+```console
+$ docker stop sonarqube
+$ docker rm sonarqube
+```  
+
+6. Run docker:
+```console
+$ docker run -d --name sq -p 9000:9000 \
+    -v $SQ_HOME/conf:/opt/sonarqube/conf \
+    -v $SQ_HOME/extensions:/opt/sonarqube/extensions \
+    -v $SQ_HOME/logs:/opt/sonarqube/logs \
+    -v $SQ_HOME/data:/opt/sonarqube/data \
+    sonarqube
+```
 
-### Option 2: Use bind-mounted persistent volumes
+Other option : use a symlink instead of an env. variable
 
-The images contain the SonarQube installation at `/opt/sonarqube`. You can use bind-mounted persistent volumes to override selected files or directories, for example:
+## Advanced configuration
 
--	`sonarqube_conf:/opt/sonarqube/conf`: configuration files, such as `sonar.properties`
--	`sonarqube_data:/opt/sonarqube/data`: data files, such as the embedded H2 database and Elasticsearch indexes
--	`sonarqube_logs:/opt/sonarqube/logs`
--	`sonarqube_extensions:/opt/sonarqube/extensions`: plugins, such as language analyzers
+#### Use parameters via Docker environment variables
 
-You could also use bind-mounted configurations specified on the command line, for example:
+The production database can be configured with the following SonarQube properties used as environment variables: `sonar.jdbc.username`, `sonar.jdbc.password` and `sonar.jdbc.url`.
 
 ```console
 $ docker run -d --name sonarqube \
     -p 9000:9000 \
-    -v /path/to/conf:/opt/sonarqube/conf \
-    -v /path/to/data:/opt/sonarqube/data \
-    -v /path/to/logs:/opt/sonarqube/logs \
-    -v /path/to/extensions:/opt/sonarqube/extensions \
+    -e sonar.jdbc.username=sonar \
+    -e sonar.jdbc.password=sonar \
+    -e sonar.jdbc.url=jdbc:postgresql://localhost/sonar \
+    -v $SQ_HOME/conf:/opt/sonarqube/conf \
+    -v $SQ_HOME/extensions:/opt/sonarqube/extensions \
+    -v $SQ_HOME/logs:/opt/sonarqube/logs \
+    -v $SQ_HOME/data:/opt/sonarqube/data \
     sonarqube
 ```
 
-### Option 3: Customized image
+Use of the environment variables `SONARQUBE_JDBC_USERNAME`, `SONARQUBE_JDBC_PASSWORD`, and `SONARQUBE_JDBC_URL` is deprecated and will stop working in future releases.
+
+More recipes can be found [here](https://github.com/SonarSource/docker-sonarqube/blob/master/recipes.md).
+
+You can pass `sonar.` configuration properties as Docker environment variables, as demonstrated in the example above for database configuration.
```

Comment:
That is means: You can pass **any** `sonar.` configuration properties?

Reply:
I'm removing this, this is not true